com.unity.multiplayer.mlapi/Runtime/Core/MessageBatcher.cs

```diff
@@ -0,0 +1,198 @@
+using MLAPI.Serialization.Pooled;
+using MLAPI.Serialization;
+using MLAPI.Configuration;
+using MLAPI.Profiling;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.IO;
+
+namespace MLAPI
+{
+    class MessageBatcher
+    {
+        public class SendStream
+        {
+            public FrameQueueItem Item;
+            public PooledBitStream Stream = new PooledBitStream();
+        }
+
+        // Stores the stream of batched RPC to send to each client, by ClientId
+        private Dictionary<ulong, SendStream> SendDict = new Dictionary<ulong, SendStream>();
+        private PooledBitWriter Writer = new PooledBitWriter(new PooledBitStream());
+
+        // Used to store targets, internally
+        private List<ulong> TargetList = new List<ulong>();
+
+        private void PushLength(int length, ref PooledBitWriter writer)
+        {
+            // If length is single byte we write it
+            if (length < 256)
+            {
+                writer.WriteByte((byte)length); // write the amounts of bytes that are coming up
+            }
+            else
+            {
+                // otherwise we write a 0, and a two-byte length
+                writer.WriteByte(0); // mark larger size
```

Comment:
we might make this marker value a `const byte` variable rather than magical `0`

Reply:
Yeah, good point, for a marker, having a `const byte` definitely makes sense. I'll add it in.